Source/DafnyCore/Backends/Cplusplus/CppCodeGenerator.cs

```diff
@@ -226,9 +226,18 @@ private string InstantiateTemplate(List<Type> typeArgs) {
 
     protected override string GetHelperModuleName() => "_dafny";
 
-    protected override IClassWriter CreateClass(string moduleName, string name, bool isExtern, string/*?*/ fullPrintName, List<TypeParameter>/*?*/ typeParameters, TopLevelDecl cls, List<Type>/*?*/ superClasses, IOrigin tok, ConcreteSyntaxTree wr) {
+    private string clName(TopLevelDecl cl) {
+      var className = IdName(cl);
+      if (cl is ClassDecl || cl is DefaultClassDecl) {
+        return className;
+      }
+      return "class_" + className;
```

Comment:
```suggestion
      return "_class_" + className;
```
That way we are sure we don't clash with something named "class_..." already in the dafny code.

Reply:
The "class_" + className encoding was the one already done by the CppCodeGenerator. I'm not sure I want to take the risk to modify it in this change, maybe a subsequent one?

Reply:
Ok, let's postpone it for a subsequent one, perhaps either proactively or if someone hits this issue. It's not a soundness issue anyway, just annoying if it happened, but it's unlikely in practice.